datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -2192,7 +2193,7 @@ fn substrait_field_ref(index: usize) -> Result<Expression> {
                     }),
                 )),
             })),
-            root_type: None,
+            root_type: Some(RootType::RootReference(RootReference{}) ),
```

Comment:
Not necessary, but if you feel like it, could remove the duplication here: split this function into `substrait_field_ref` that returns the FieldReference and `substrait_field_ref_expr` that wraps it into Expression and then reusing `substrait_field_ref` in the `try_to_substrait_field_reference`.

Reply:
Yes - I agree; though my rust is a few years out of date.. so getting up to speed again. So trying to make as few changes as needed...